pkg/utils/release/controller.go

```diff
@@ -424,3 +450,132 @@ func (s *SuiteController) CreateReleasePipelineRoleBindingForServiceAccount(name
 	}
 	return roleBinding, nil
 }
+
+func (s *SuiteController) WaitForReleasePipelineToBeFinished(managedNamespace, devNamespace string, releaseName string, component *appstudioApi.Component, maxRetries int) error {
+	attempts := 1
+	var pr *v1beta1.PipelineRun
+	var prList *v1beta1.PipelineRunList
+
+	for {
+		err := wait.PollImmediate(20*time.Second, 30*time.Minute, func() (done bool, err error) {
+
+			if len(releaseName) > 0 {
+				prList, err = s.GetPipelineRunListInNamespace(managedNamespace, releaseName, devNamespace)
+				if err != nil {
+					GinkgoWriter.Println("PipelineRun has not been created yet.")
+					return false, nil
+				}
+			} else {
+				prList, err = s.GetAllReleasePipelineRunMatchComponent(managedNamespace, devNamespace, component)
+				if err != nil {
+					GinkgoWriter.Println("PipelineRun has not been created yet.")
+					return false, nil
+				}
+
+			}
+
+			pr := &prList.Items[0]
+			GinkgoWriter.Printf("PipelineRun %s reason: %s\n", pr.Name, pr.GetStatusCondition().GetCondition(apis.ConditionSucceeded).GetReason())
+
+			if !pr.IsDone() {
+				return false, nil
+			}
+
+			if pr.GetStatusCondition().GetCondition(apis.ConditionSucceeded).IsTrue() {
+				return true, nil
+			} else {
+				var prLogs string
+				if err = tekton.StorePipelineRun(pr, s.KubeRest(), s.KubeInterface()); err != nil {
+					GinkgoWriter.Printf("failed to store PipelineRun %s:%s: %s\n", pr.GetNamespace(), pr.GetName(), err.Error())
+				}
+				if prLogs, err = tekton.GetFailedPipelineRunLogs(s.KubeRest(), s.KubeInterface(), pr); err != nil {
+					GinkgoWriter.Printf("failed to get logs for PipelineRun %s:%s: %s\n", pr.GetNamespace(), pr.GetName(), err.Error())
+				}
+				return false, fmt.Errorf(prLogs)
+			}
+		})
+
+		if err != nil {
+			GinkgoWriter.Printf("attempt %d/%d: PipelineRun %q failed: %+v", attempts, maxRetries+1, pr.GetName(), err)
+			attempts++
+		} else {
+			break
+		}
+	}
+
+	return nil
+
+}
+
+// GetAllReleasePipelineRunMatchComponent returns the release pipelineRuns in namespace associated with component.
+func (s *SuiteController) GetAllReleasePipelineRunMatchComponent(managedNamespace, devNamespace string, component *appstudioApi.Component) (*v1beta1.PipelineRunList, error) {
+
+	appName := component.Spec.Application
```

Comment:
this is only used in line 525. Should probably just use `component.Spec.Application` there and not define this

Reply:
you are right, 
removed and used `component.Spec.Application` as argument